iris-machine-learning-tutorial/README.md

```diff
@@ -0,0 +1,103 @@
+# Tutorial: Using Amazon Lambda and ECS to predict Iris species using Scikit-Learn
+
+In this tutorial, you’ll learn how to create a serverless Lambda function, deployed as a container image, to predict Iris species using Scikit-Learn. This tutorial is intended for Pro users who wish to get more acquainted with Lambda, ECS, and LocalStack's Cloud Pods for reproducible machine learning.
+
+## Prerequisites
+
+* LocalStack
+* Docker
+* `make`
+* `awslocal`
+
+## Installing
+
+To install the dependencies:
+
+```sh
+make install
+```
+
+## Create a container image
+
+Create an ECR repository:
+
+```sh
+awslocal ecr create-repository --repository-name iris-machine-learning-tutorial | jq -r '.repository.repositoryUri'
+```
+
+Build the container image:
+
+```sh
+docker build -t localhost:4510/iris-machine-learning-tutorial .
+```
+
+Push the container image to the repository:
+
+```sh
+docker push localhost:4510/iris-machine-learning-tutorial:latest
+```
+
+## Create the Lambda function
+
+Create the Lambda function:
+
+```sh
+awslocal lambda create-function --function-name iris-prediction --package-type Image \
+		--code ImageUri=localhost:4510/repo1 --role arn:aws:iam::000000000:role/lambda-r1 \
```

Comment:
Same as above:
- Where is this repo1 coming from?
- Would we get an error if the IAM role is not there?

Reply:
> Where is this repo1 coming from?

Answered above. It was for test which I have removed.

> Would we get an error if the IAM role is not there?

Yes

Reply:
Ok, the IAM role doesn't result in an error in LocalStack. Id does result in one in AWS:
`An error occurred (ValidationException) when calling the CreateFunction operation: 1 validation error detected: Value 'arn:aws:iam::000000000:role/lambda-r1' at 'role' failed to satisfy constraint: Member must satisfy regular expression pattern: arn:(aws[a-zA-Z-]*)?:iam::\d{12}:role/?[a-zA-Z_0-9+=,.@\-_/]+`
Even if I change 9 zeros to 12 zeros - I gate another errors:
`An error occurred (AccessDeniedException) when calling the CreateFunction operation: Cross-account pass role is not allowed.`
Which is expected, since the account ID doesn't match the one in AWS.

What I am worried about here - is that we are giving our customers a sequence of commands that works in LocalStack, but doesn't in AWS. As such, we fail as a "learning tool for AWS".

Would be great to be able to use the same tutorial in AWS and in LocalStack. If creation of a required IAM role is not difficult - would be great to add it to this tutorial. But at the very least - would be useful to add a comment on this. Something along the lines of "In LocalStack Lambda doesn't enforce IAM roles, so we can supply a non-existent role as a parameter, making this tutorial simpler. In AWS, though, you have to create a valid IAM role for your lambda first".